x/ibc/03-connection/types/connection.go

```diff
@@ -72,6 +83,12 @@ func (c ConnectionEnd) ValidateBasic() error {
 			return sdkerrors.Wrap(ibctypes.ErrInvalidVersion, "version can't be blank")
 		}
 	}
+	if c.ID == c.Counterparty.ConnectionID {
```

Comment:
I added these 2 extra validations @cwgoes

Reply:
Why? It's fine if the identifiers are the same.